src/lib/io/sol-iio.c

```diff
@@ -329,11 +380,19 @@ check_trigger(struct sol_iio_device *device)
     }
 
     len = sol_util_read_file(path, "%ms", &name);
+
     if (len < 0) {
-        SOL_INF("No current trigger for iio:device%d. Creating a sysfs one.",
-            device->device_id);
-        if (!create_sysfs_trigger(device))
-            return false;
+        if (streqn(config->trigger_name, HRTIMER_TRIGGER_PREFIX, strlen(HRTIMER_TRIGGER_PREFIX))) {
```

Comment:
Remember that Soletta has a `strstartswith` function, please use it as it better explain code intent (check if a string starts with some prefix).